FoodplannerApi/Controller/ImagesController.cs

```diff
@@ -58,18 +68,59 @@ public async Task<IActionResult> DeleteImages(IEnumerable<int> foodImageIds)
     }
 
     [HttpGet]
+    [Authorize(Roles = "Children, Parent")]
+    [ServiceFilter(typeof(AuthoriseImageOwnerFilter))]
     public async Task<IActionResult> GetFoodImage(int foodImageId)
     {
         if (foodImageId < 0)
-            return BadRequest("Invalid userId provided");
-        var foodImage = await foodImageService.GetFoodImage(foodImageId);
-        return Ok(new { foodImage });
+            return BadRequest("Invalid userId");
+        return Ok(foodImageService.GetFoodImage(foodImageId));
     }
 
     [HttpGet]
+    [Authorize(Roles = "Children, Parent")]
+    [ServiceFilter(typeof(AuthoriseImageOwnerFilter))]
+
     public async Task<IActionResult> GetPresignedImageLink(int foodImageId)
     {
         var presignedImageLink = await foodImageService.GetFoodImageLink(foodImageId);
         return Ok(presignedImageLink);
     }
+
+    public class AuthoriseImageOwnerFilter : IAsyncActionFilter
+    {
+        public async Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+        {
+            var authService = context.HttpContext.RequestServices.GetService<AuthService>();
+            var foodImageService = context.HttpContext.RequestServices.GetService<IFoodImageService>();
+            var token = context.HttpContext.Request.Headers["Authorization"].FirstOrDefault();
+            var foodImageIds = context.HttpContext.Request.Query["foodImageId"];
+            if (token == null)
+            {
+                context.Result = new UnauthorizedResult();
+            } else if (foodImageIds.Any(id => id == null))
+            {
+                context.Result = new BadRequestResult();
+            }
+            else if (authService == null || foodImageService == null)
+            {
+                throw new Exception("Missing services");
+            } else
+            {
+                var userid = authService.RetrieveIdFromJWTToken(token);
+                foreach (var foodImageId in foodImageIds)
+                {
+                    var foodImage = await foodImageService.GetFoodImage(int.Parse(foodImageId));
+                    if (userid == foodImage.UserId.ToString()) continue;
+                    context.Result = new UnauthorizedResult();
+                    break;
+                }
+            }
+        }
+
+        public void OnActionExecuted(ActionExecutedContext context)
+        {
+            throw new NotImplementedException();
+        }
```

Comment:
Skal denne bruges til noget ?

Reply:
Den er nødvendig for at implementere IAsyncActionFilter interfacet, men vi skal ikke bruge den til noget nej. Med det sagt ville det nok være smartere bare at returnere i stedet for at kaste en exception

Reply:
Nvm. den kunne åbenbart godt fjernes alligevel, så det er hermed gjort 👍